File: desktop-mend/src/test/java/commands/CleanTest.java
package commands;

import co.samco.mend4.core.exception.CorruptSettingsException;
import co.samco.mend4.core.Settings;
import co.samco.mend4.desktop.commands.Clean;
import co.samco.mend4.desktop.exception.SettingRequiredException;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;
import java.util.Collections;

import static org.mockito.Mockito.*;

public class CleanTest extends TestBase {
    private Clean clean;

    @Before
    public void setup() {
        super.setup();
        clean = new Clean(strings, log, settingsHelper, shredHelper, settings);
    }

    @Test
    public void testCommandWithSettingsDependencies() throws IOException, SettingRequiredException {
        super.testCommandWithSettingsDependencies(clean);
    }

    @Test
    public void executesShredCommand() throws IOException, CorruptSettingsException {
        String decDir = "dir";
        when(settings.getValue(eq(Settings.Name.DECDIR))).thenReturn(decDir);
        clean.execute(Collections.emptyList());
        verify(shredHelper).shredFilesInDirectory(eq(decDir));
        verify(err).println(eq(strings.get("Clean.cleanComplete")));
    }

    @Test
    public void printsException() throws CorruptSettingsException, IOException {
        String exception = "hi";
        when(settings.getValue(eq(Settings.Name.DECDIR))).thenThrow(new CorruptSettingsException(exception, ""));
        clean.execute(Collections.emptyList());
        verify(err).println(eq(exception));
    }
}

File: desktop-mend/src/test/java/commands/CleanTest.kt
package commands

import co.samco.mend4.desktop.commands.Clean
import co.samco.mend4.desktop.dao.SettingsDao
import com.nhaarman.mockitokotlin2.any
import org.junit.Before
import org.junit.Test
import com.nhaarman.mockitokotlin2.eq
import com.nhaarman.mockitokotlin2.verify
import com.nhaarman.mockitokotlin2.whenever
import java.io.File

class CleanTest : TestBase() {
    private lateinit var clean: Clean

    @Before
    override fun setup() {
        super.setup()
        clean = Clean(
            strings = strings,
            log = log,
            settingsHelper = settingsHelper,
            shredHelper = shredHelper,
            settings = settings
        )
    }

    @Test
    fun testCommandWithSettingsDependencies() {
        super.testCommandWithNoSettingsDependencies(clean)
    }

    @Test
    fun executesShredCommand() {
        val decDir = "dir"
        whenever(settings.getValue(eq(SettingsDao.DEC_DIR))).thenReturn(decDir)
        whenever(settings.getValue(eq(SettingsDao.SHRED_COMMAND))).thenReturn(" ")
        whenever(osDao.listFiles(any())).thenReturn(arrayOf(File("test-file")))

        clean.execute(emptyList())
        verify(shredHelper).shredFilesInDirectory(eq(decDir))
        verify(err).println(eq(strings["Clean.cleanComplete"]))
    }

    @Test
    fun printsException() {
        whenever(settings.getValue(eq(SettingsDao.DEC_DIR))).thenReturn(null)
        clean.execute(emptyList())
        verify(err).println(
            eq(
                strings.getf(
                    "General.propertyNotSet",
                    SettingsDao.DEC_DIR.encodedName,
                    Clean.COMMAND_NAME
                )
            )
        )
    }
}